src/d/actor/d_a_obj_doguu.cpp

```diff
@@ -5,120 +5,811 @@
+    JUT_ASSERT(0x18f, modelData != NULL);
+    bmt =  static_cast<J3DMaterialTable *>(dComIfG_getObjectRes("Doguu", daObjDoguu_idx_table.bmt_vgsb[field_0x894]));
+    modelData->setMaterialTable(bmt, J3DMatCopyFlag_All);
+    field_0x6D4 = mDoExt_J3DModel__create(modelData, 0x80000,0x15220202);
+    
+    if (field_0x6D4 == NULL) {
+      return FALSE;
+    }
+    
+    JUTNameTab* nameTable = field_0x6D4->getModelData()->getJointName();
+    
+    for (u16 i = 0; i < field_0x6D4->getModelData()->getJointNum(); i++) {
+        const char* name = nameTable->getName(i);
+        if (strcmp("head1", name) == 0) {
+            mJointIdx = i;
+            break;
+        }
+    }
+    
+    J3DAnmTransform* bck_body = static_cast<J3DAnmTransform *>(dComIfG_getObjectRes("Doguu", 6));
+    JUT_ASSERT(0x1AB, bck_body != NULL);
+    
+    if (mBck2.init(modelData, bck_body, true, J3DFrameCtrl::EMode_NONE, 1.0f, 0, -1, false) == 0) {
+      return FALSE;
+    }
+
+    deform = new J3DSkinDeform();
+
+    if(deform) {
+        field_0x6D4->setSkinDeform(deform, 1);
+    } else {
+        return FALSE;
+    }
+
+    modelData = (J3DModelData*)(dComIfG_getObjectRes("Doguu", daObjDoguu_idx_table.bdl_vgsp[field_0x894]));
+    JUT_ASSERT(0x1bd, modelData != NULL);
+    field_0x6D8 = mDoExt_J3DModel__create(modelData, 0,0x11020203);
+
+    if (field_0x6D8 == NULL) {
+        return FALSE;
+    }
+
+    J3DAnmTransform* bck_crystal = static_cast<J3DAnmTransform*>(dComIfG_getObjectRes("Doguu", DOGUU_BCK_VGSPA));
+    JUT_ASSERT(0x1c3, bck_crystal != NULL);
+    
+    if(mBck3.init(modelData, bck_crystal, true, J3DFrameCtrl::EMode_NONE, 1.0f, 0, -1, false) == 0){
+        return FALSE;
+    } else{
+        field_0x89C = 1;
+        return TRUE;
+    }
 }
 
 /* 000007D0-00000B80       .text CreateInit__12daObjDoguu_cFv */
 void daObjDoguu_c::CreateInit() {
-    /* Nonmatching */
+    
+
+    cullMtx = field_0x6CC->getBaseTRMtx();
+    fopAcM_setCullSizeBox(this, -600.0f, -0.0f, -600.0f, 600.0f, 400.0f, 600.0f);
+    fopAcM_setCullSizeFar(this, 1.0f);
+    if (field_0x894 == 0){
+        mItemNo = dItem_PEARL_DIN_e;
+    } else if(field_0x894 == 1) {
+        mItemNo = dItem_PEARL_FARORE_e;
+    } else{
+        mItemNo = dItem_PEARL_NAYRU_e;
+    }
+    mStts.Init(0xff, 0xff, this);
+    mCyl.Set(l_cyl_src);
+    if(!checkItemGet(mItemNo, 1)){
+        mCyl.SetR(50.f);
+    }
+    mCyl.SetStts(&mStts);
+    dKy_plight_set(&mLightInfluence);
+    if (subtype >= 1) {
+        mBck1.setFrame(mBck1.getStartFrame());
+        mBck2.setFrame(mBck2.getStartFrame());
+        mBck3.setFrame(mBck3.getStartFrame());
+
+        field_0x8A0 = true;
+        field_0x8A1 = false;
+        field_0x8A2 = true;
+        field_0x8AC = 14;
+    } else {
+        int eventBit = dComIfGs_isEventBit(0x1480);
+        if ((eventBit != false) && (eventBit = dComIfGs_isEventBit(0x1440),
+            eventBit != 0) && (eventBit = dComIfGs_isEventBit(0x1410),
+            eventBit != 0)) {
```

Comment:
dComIfGs_isEventBit doesn't return an eventBit, it checks if an eventBit is set. Also, assignments within an if statement is a Ghidra thing and should be avoided unless strictly necessary. Try something like this:
```suggestion
    } else if (
        dComIfGs_isEventBit(0x1480) &&
        dComIfGs_isEventBit(0x1440) &&
        dComIfGs_isEventBit(0x1410)
    ) {
```